### PR TITLE
MM-22355 - Fix typos detected by goreportcad.com

### PR DESCRIPTION
**Pull request by JtheBAB**
#### Summary
Fixes the typos found by goreportcad.com

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/13835
Jira: https://mattermost.atlassian.net/browse/MM-22355

### CODE REVIEW

File: utils/markdown/html_entities.go [hunk 1]
```diff
@@ -1145,7 +1145,7 @@ var htmlEntities = map[string]string{
 	"in":                              "\u2208",
 	"incare":                          "\u2105",
 	"infin":                           "\u221E",
-	"infintie":                        "\u29DD",
+	"infinite":                        "\u29DD",
```
Comment (JtheBAB) [inline utils/markdown/html_entities.go]: Not sure

**Comment by JtheBAB** [comment]
I guess [this](https://github.com/mattermost/mattermost-server/blob/master/utils/markdown/html_entities.go#L625) is a false positive

**Comment by jespino** [comment]
> I guess [this](https://github.com/mattermost/mattermost-server/blob/master/utils/markdown/html_entities.go#L625) is a false positive

Yes, I think so :)

File: utils/markdown/html_entities.go (hunk 1 above)
Comment (jespino) [reply utils/markdown/html_entities.go]: False positive, take a look here: https://www.w3schools.com/charsets/ref_html_entities_i.asp and search for the infintie html entity.

**Comment by JtheBAB** [comment]
:/

Google fooled me. Couldn’t find the /code term.

**Comment by hanzei** [comment]
@justinegeffen could you please review this PR? I don't feel comfortable reviewing it as a non native speaker.

File: app/post_metadata.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func (a *App) OverrideIconURLIfEmoji(post *model.Post) {
 	if emojiUrl, err := a.GetEmojiStaticUrl(emojiName); err == nil {
 		post.AddProp(model.POST_PROPS_OVERRIDE_ICON_URL, emojiUrl)
 	} else {
-		mlog.Warn("Failed to retrieve URL for overriden profile icon (emoji)", mlog.String("emojiName", emojiName), mlog.Err(err))
+		mlog.Warn("Failed to retrieve URL for overridden profile icon (emoji)", mlog.String("emojiName", emojiName), mlog.Err(err))
```
Comment (justinegeffen) [inline app/post_metadata.go]: Should this be "overwritten"?

Comment (agnivade) [reply app/post_metadata.go]: I think overridden is the correct form. The function itself is called OverrideIconURL and it does exactly that - override the emoji.

**Comment by jasonblais** [comment]
@jespino Looks like this one is ready to merge

**Pull request merged**